### PR TITLE
feat(suite): format crypto amounts based on app locale

### LINKED ISSUE #5545: Extend thousands separator to crypto currencies
Thousand separators are not set for crypto currencies - only for fiat.
Extend it for cryptocurrencies as well.

Visible with Sats units on Bitcoin accounts (or on Doge coin for example)
Related to: https://github.com/trezor/trezor-suite/issues/3004

### PR DESCRIPTION
**Pull request by dahaca**
closes [#5545](https://github.com/trezor/trezor-suite/issues/5545)

### CODE REVIEW

File: packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx [hunk 1]
```diff
@@ -52,23 +57,23 @@ export const FormattedCryptoAmount = ({
...
-        formattedValue = formatCurrencyAmount(
-            Number(networkAmountToSatoshi(String(value), symbol as NetworkSymbol)),
-        ) as string;
+        formattedValue = Number(networkAmountToSatoshi(String(value), symbol as NetworkSymbol));
 
         formattedSymbol = isTestnet ? `sat ${symbol?.toUpperCase()}` : 'sat';
     }
 
     if (isRawString) {
         return (
```
Comment (0ex-d) [inline packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]:
```suggestion
    if (isRawString) {
    const displaySignValue = signValue ? `${isValuePositive(signValue) ? '+' : '-'}` : '';
        return (
```

File: packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx [hunk 2]
```diff
@@ -52,23 +57,23 @@ export const FormattedCryptoAmount = ({
...
             <>
-                {`${signValue ? `${isValuePositive(signValue) ? '+' : '-'}` : ''} ${formattedValue}
+                {`${
+                    signValue ? `${isValuePositive(signValue) ? '+' : '-'}` : ''
+                } ${formatCurrencyAmount(formattedValue, locale)}
                 ${formattedSymbol}`}
```
Comment (0ex-d) [inline packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]: A little hard to read, can we move some of the if condition out side the render function. e.g
```suggestion
               {displaySignValue} {formatCurrencyAmount(formattedValue, locale)} {formattedSymbol}
```

**Review by 0ex-d — COMMENTED** [review]
Looks good

File: packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx [hunk 3]
```diff
@@ -52,23 +57,23 @@ export const FormattedCryptoAmount = ({
 
     const areSatsSupported = !!networkFeatures?.includes('amount-unit');
 
-    let formattedValue = value;
+    let formattedValue = Number(value);
```
Comment (matejkriz) [inline packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]: the name is confusing, it is not formatted value, right? it's more like parsed value

Comment (dahaca) [reply packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]: Named it that way because it can also be converted to satoshis later. So it's not just parsed :( What name would you prefer?

Comment (matejkriz) [reply packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]: There is conversion to Number even if isSatoshis, so maybe `valueNumber`?

**Review by matejkriz — CHANGES_REQUESTED** [review]
EDIT: fixed

There are more places that needs to be touched, because we pass formatted string values on some places and it leads to `NaN` values displayed
image

File: packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx (hunk 1 above)
Comment (dahaca) [reply packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]: 3aa96a54055bc0b98fa0be83e3e90e6936d51552

File: packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx (hunk 2 above)
Comment (dahaca) [reply packages/suite/src/components/suite/FormattedCryptoAmount/index.tsx]: 3aa96a54055bc0b98fa0be83e3e90e6936d51552

File: suite-common/wallet-utils/src/__tests__/balanceUtils.test.ts [hunk 4]
```diff
@@ -13,16 +13,15 @@ test('formatBalanceUtils', () => {
...
+    expect(formatCoinBalance('1000.0001')).toEqual('1,000.0001');
+    expect(formatCoinBalance('1000.000000001')).toEqual('1,000.00');
+    expect(formatCoinBalance('2600.1')).toEqual('2,600.1');
+    expect(formatCoinBalance('200000')).toEqual('200,000');
+    expect(formatCoinBalance('2000000')).toEqual('2,000,000');
+    expect(formatCoinBalance('2900000')).toEqual('2,900,000');
+    expect(formatCoinBalance('0099999999.999999999999')).toEqual('99,999,999.9…');
     expect(formatCoinBalance('0.12345678')).toEqual('0.12345678');
     expect(formatCoinBalance('10.12345678')).toEqual('10.1234567…');
+    expect(formatCoinBalance('10.12345678', 'cs')).toEqual('10,1234567…');
```
Comment (matejkriz) [inline suite-common/wallet-utils/src/__tests__/balanceUtils.test.ts]: I would suggest to add more cases for all the locales available in Suite

File: suite-common/wallet-utils/src/numberFormatUtils.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export const getDecimalSeparator = (locale: string) => {
```
Comment (matejkriz) [inline suite-common/wallet-utils/src/numberFormatUtils.ts]: those utils are not used or am I looking wrong?

Comment (dahaca) [reply suite-common/wallet-utils/src/numberFormatUtils.ts]: Actually, yes, I was trying one thing but ended up not needing them. Will remove :)

File: packages/suite/src/components/suite/FormattedCryptoAmount.tsx [hunk 6]
```diff
@@ -57,21 +66,25 @@ export const FormattedCryptoAmount = ({
...
+        formattedValue = networkAmountToSatoshi(String(value), symbol as NetworkSymbol);
 
         formattedSymbol = isTestnet ? `sat ${symbol?.toUpperCase()}` : 'sat';
     }
 
+    // format truncation + locale (used for balances) or just locale
+    if (isBalance) {
+        formattedValue = formatCoinBalance(String(formattedValue), locale);
+    } else {
+        formattedValue = formatCurrencyAmount(Number(formattedValue), locale);
```
Comment (tomasklim) [inline packages/suite/src/components/suite/FormattedCryptoAmount.tsx]:
```suggestion
        formattedValue = formatNumber(Number(formattedValue), locale);
```

Nit:
Currency amount in formatted crypto amount does not make sense

File: suite-common/wallet-utils/src/balanceUtils.ts [hunk 7]
```diff
@@ -25,11 +24,13 @@ export const formatCoinBalance = (value: string) => {
...
         const noDecimalsLeft = fixedBalanceBig.modulo(2).toFixed() === '0';
         if (noDecimalsLeft) {
-            return fixedBalanceBig.toFixed(2);
+            return formatCurrencyAmount(fixedBalanceBig.toNumber(), locale, 2);
         }
 
-        return isTruncated ? `${fixedBalanceBig.toFixed()}…` : fixedBalanceBig.toFixed();
+        const localizedBalance = formatCurrencyAmount(fixedBalanceBig.toNumber(), locale);
+
+        return isTruncated ? `${localizedBalance}…` : localizedBalance;
```
Comment (tomasklim) [inline suite-common/wallet-utils/src/balanceUtils.ts:32]: Would be nice to show tooltip with the whole amount in some follow-up issue.

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx [hunk 8]
```diff
@@ -138,7 +138,7 @@ const Amount = ({ output, outputId }: Props) => {
     }
 
     const withTokens = hasNetworkFeatures(account, 'tokens');
-    const symbolToUse = areSatsDisplayed ? 'sats' : symbol.toUpperCase();
+    const symbolToUse = areSatsDisplayed ? 'sat' : symbol.toUpperCase();
```
Comment (matejkriz) [inline packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx]: This whole commit could be in separate PR e27460095ef5a95cf7a47763f8070a441d89efd6

File: packages/suite/src/components/suite/FormattedCryptoAmount.tsx [hunk 9]
```diff
@@ -35,12 +41,15 @@ interface FormattedCryptoAmountProps {
 export const FormattedCryptoAmount = ({
     value,
     symbol,
+    isBalance,
```
Comment (matejkriz) [inline packages/suite/src/components/suite/FormattedCryptoAmount.tsx:44]: In general, I would prefer composition over configuration so I always try to avoid those switches like `isBalance` and `isRawString` and `disableHiddenPlaceholder`, but I find that tolerable here.

Comment (dahaca) [reply packages/suite/src/components/suite/FormattedCryptoAmount.tsx:44]: How would you write it? :)

File: suite-common/wallet-utils/src/balanceUtils.ts (hunk 7 above)
Comment (dahaca) [reply suite-common/wallet-utils/src/balanceUtils.ts:32]: I think we should discuss it later again together with @hynek-jina and @sime

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx (hunk 8 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx]: https://github.com/trezor/trezor-suite/pull/5923

File: packages/suite/src/components/suite/FormattedCryptoAmount.tsx (hunk 6 above)
Comment (dahaca) [reply packages/suite/src/components/suite/FormattedCryptoAmount.tsx]: 97bf0112f9b74c51f4c29d88982cf9e0e3388d9c

File: suite-common/wallet-utils/src/numberFormatUtils.ts (hunk 5 above)
Comment (dahaca) [reply suite-common/wallet-utils/src/numberFormatUtils.ts]: 97bf0112f9b74c51f4c29d88982cf9e0e3388d9c

File: suite-common/wallet-utils/src/__tests__/balanceUtils.test.ts (hunk 4 above)
Comment (dahaca) [reply suite-common/wallet-utils/src/__tests__/balanceUtils.test.ts]: 97bf0112f9b74c51f4c29d88982cf9e0e3388d9c

File: packages/suite/src/components/suite/FormattedCryptoAmount.tsx (hunk 9 above)
Comment (matejkriz) [reply packages/suite/src/components/suite/FormattedCryptoAmount.tsx:44]: it's possible to split it to multiple components that would share what is common and handle only the differences. So maybe creating some general component that could receive the formatter from more specialised components like FormattedBalance, FormattedRawString, ... but as I said, I'm ok with the current solution here but we should avoid to add more config props to this component in the future.

**Review by matejkriz — APPROVED** [review]

**Review by tomasklim — APPROVED** [review]
